src/segmented-control/segmented-control.scss

```diff
@@ -0,0 +1,173 @@
+  transition: border-color, background-color 180ms cubic-bezier(0.3, 0.1, 0.5, 1);
+  align-items: center;
+  justify-content: center;
+  gap: var(--primer-control-medium-gap, $spacer-2);
+
+  &:hover {
+    background-color: var(--color-segmented-control-hover-bg);
+    transition-duration: 120ms;
+  }
+
+  &:active {
+    background-color: var(--color-segmented-control-active-bg);
+    border-color: var(--color-segmented-control-active-border);
+    transition-duration: 60ms;
+  }
+
+  // Pressed
+
+  // TODO: Change to aria-current or aria-selected?
+  &[aria-pressed='true'] {
+    // stylelint-disable-next-line primer/typography
+    font-weight: var(--base-text-weight-semibold, $font-weight-bold);
+    background-color: var(--color-segmented-control-pressed-bg);
+    border-color: var(--color-segmented-control-pressed-border);
+    box-shadow: var(--color-shadow-medium);
+  }
+
+  // Divider
+
+  // stylelint-disable-next-line scss/selector-no-redundant-nesting-selector
+  & + .SegmentedControl-button::before {
+    position: absolute;
+    top: 1px; // Account for border width
+    left: -1px;
+    height: var(--primer-text-body-size-large, 16px);
+    // stylelint-disable-next-line primer/spacing
+    margin-top: var(--primer-control-medium-paddingBlock, 6px);
+    content: '';
+    // stylelint-disable-next-line primer/borders
+    border-left: var(--primer-borderWidth-thin, $border-width) $border-style var(--color-border-default);
+    transition: border-color 180ms cubic-bezier(0.3, 0.1, 0.5, 1);
+  }
+
+  // Remove dividers
+
+  &[aria-pressed='true'] + .SegmentedControl-button::before {
+    border-color: transparent;
+  }
+
+  &:hover,
+  &:active,
+  &:focus-visible {
+    &::before,
+    + .SegmentedControl-button::before {
+      border-color: transparent;
+      transition-duration: 60ms;
+    }
+  }
+}
+
+// Icon -----------------------------------------
+
+.SegmentedControl-leadingIcon {
+  color: var(--color-fg-muted);
+}
+
+// Label -----------------------------------------
+
+.SegmentedControl-label {
+  // renders a visibly hidden "copy" of the label in bold, reserving box space for when label becomes bold on selected
+  &[data-content]::before {
+    display: block;
+    height: 0;
+    // stylelint-disable-next-line primer/typography
+    font-weight: var(--base-text-weight-semibold, $font-weight-bold);
+    visibility: hidden;
+    content: attr(data-content);
+  }
+}
+
+// Variants -----------------------------------------
+
+// disabled
+.SegmentedControl--disabled { // TODO: Replace with aria-disabled?
+  .SegmentedControl-button {
+    color: var(--color-primer-fg-disabled);
+    pointer-events: none;
+    cursor: default;
+
+    &[aria-pressed='true'] {
+      border-color: var(--color-border-default);
+    }
+  }
+
+  .SegmentedControl-leadingIcon {
+    color: var(--color-primer-fg-disabled);
+  }
+}
+
+.SegmentedControl-button--isLoading.SegmentedControl-button--isLoading {
```

Comment:
Currently each button can have their own loading state when pressed and a spinner is shown inside the button:

<img width="149" alt="image" src="https://user-images.githubusercontent.com/378023/171609304-4657da03-7ed7-4002-b7c1-be0b2949664a.png">

An alternative would be to show a spinner outside the `SegmentedControl ` as seen in [Figma](https://www.figma.com/file/vEbF2YUD0lRrl5Mu4z7Gx3/Segmented-control---FY22-Q4?node-id=42%3A14069):

<img width="311" alt="image" src="https://user-images.githubusercontent.com/378023/171609906-2fefb8f1-198b-48cb-9d29-0472647d7786.png">

Reply:
I've been confused about a loading state in these designs. Why do we need spinners directly inside (or attached to) this component? A temporary loading state that changes the element dimensions feels like a fragile pattern design here.

c/c @mperrotti

Reply:
Added a comment about this here: https://github.com/github/primer/issues/761#issuecomment-1145190534 (only available for hubbers)

Reply:
> A temporary loading state that changes the element dimensions feels like a fragile pattern design here.

I agree with @vdepizzol on this, but I've been unable to design a better way to indicate loading without showing a loading spinner that changes the component's width.

That being said, I prefer showing a spinner outside the `SegmentedControl`.